src/server/zset_family.cc

```diff
@@ -2030,6 +2038,43 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInter(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<SetOpArgs> op_args_res = ParseSetOpArgs(args, false);
+
+  if (!op_args_res) {
+    return HandleOpStatus(cntx, op_args_res.status());
+  }
+  const auto& op_args = *op_args_res;
+  if (op_args.num_keys == 0) {
+    return SendAtLeastOneKeyError(cntx);
+  }
+
+  vector<OpResult<ScoredMap>> maps(shard_set->size(), OpStatus::SKIPPED);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpInter(shard, t, "", op_args.agg_type, op_args.weights, false);
+    return maps[shard->shard_id()];
```

Comment:
No reason to return the map, you could just use `ScheduleSingleHop` (note the lack of `T`) and return OK

Reply:
good catch as well

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/4b9091a71c2ec0ff059a7788be4cddb6daaab902